### ISSUE TITLE
[Agent] Setting Tags/labels at enrolment or installation 

### ISSUE THREAD

**Issue by mohsen0** [elastic/beats#27712]
Ability to set labels or tags during installation or enrolment agent 
so that different identical deployments of service can be differentiated in the dataset

The specific use case is described in: https://discuss.elastic.co/t/fleet-managed-elastic-agent-environment-deployment-name-custom-field/282565/6

```
sudo ./elastic-agent install -f --url=https://<feet manager>:443 --enrollment-token=<token> --tag=staging01 
```
or in case, the agent is already installed via package manager
```
elastic-agent enroll -f --url=https://<feet manager>:443 --enrollment-token=<token> --tag=staging01 
```

**Comment by nimarezainia** [comment]
let's also provide a flag to override the agent name for more flexibility.

**Comment by mostlyjason** [comment]
I added Fleet labels here since we'd want a way to view the tags within Fleet to support easier filtering/mgmt. Optionally, we may want a way to add tags within Fleet as well, or use tags in things like processors.

**Comment by nimarezainia** [comment]
@mostlyjason, @jen-huang  do we need a separate issue for Fleet to track this?  what gets added at agent install needs to be (added) displayed at Fleet. I'm wondering how we should coordinate.

**Comment by mostlyjason** [comment]
@nimarezainia might be good to start with a product requirements and UX doc that considers the end to end user experience?

**Comment by jlind23** [comment]
@mostlyjason perfect! I've discussed with @nimarezainia about it yesterday and he had the same idea in mind.
I will put it aside until with have some clear from an engineering perspective.
@ph i'll "freeze" it for now and maybe move it to another release then.

**Comment by nimarezainia** [comment]
@jlind23 @ph @mostlyjason @jen-huang  I think most have commented on the definition document already. Help me understand how we coordinate with the Fleet team also. Do we need a separate issue there?

**Comment by jlind23** [comment]
@nimarezainia yes, we should create an issue in kibana repository and link it here as dependant.
